res/raw/webxdc.js

```diff
@@ -39,5 +39,28 @@ window.webxdc = (() => {
     sendUpdate: (payload, descr) => {
       InternalJSApi.sendStatusUpdate(JSON.stringify(payload), descr);
     },
+
+    sendToChat: (content) => {
+        var data = {};
+
+        if (content.text) {
+            data.text = content.text;
+        }
+
+        if (content.file) {
+            var file = content.file;
+            var reader = new FileReader();
+            reader.onload = function(readerEvt) {
+                var binaryString = readerEvt.target.result;
+                data.blob = btoa(binaryString);
+                data.name = file.name;
+                data.type = file.type;
+                InternalJSApi.sendToChat(JSON.stringify(data));
+            };
+            reader.readAsBinaryString(file);
+        } else {
+            InternalJSApi.sendToChat(JSON.stringify(data));
+        }
+    },
```

Comment:
```suggestion
    sendToChat: async (content) => {
        const data = {};

        if (content.text) {
            data.text = content.text;
        }

        if (content.file) {
            data.blob = await new Promise((resolve, reject) => {
                const reader = new FileReader()
                reader.onload = () => {
                  /** @type {string} */
                  let data = reader.result
                  resolve(data.slice(data.indexOf(data_start) + data_start.length))
                }
                reader.readAsDataURL(content.file)
                reader.onerror = () => reject(reader.error)
              })           
            data.name = content.file.name;
            data.type = content.file.type;
        }
        const result = InternalJSApi.sendToChat(JSON.stringify(data));
        if(!result){
            return Promise.reject("There was an error")
        }
    },
```

Reply:
how the user uses sendToChat() then? since it is the only of the existing API functions that is declared with async keyword, does that need any special treatment? or can it be normally called as any non-async function and it is ensured to be executed? without needing `await` keyword?